ndt7.go

```diff
@@ -0,0 +1,151 @@
+// Package ndt7 contains a ndt7 client.
+//
+// The client will automatically discover a suitable server to use
+// by default. However, you can also manually discover a server and
+// configure the client accordingly.
+package ndt7
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/gorilla/websocket"
+	"github.com/m-lab/ndt7-client-go/internal/download"
+	"github.com/m-lab/ndt7-client-go/internal/upload"
+	"github.com/m-lab/ndt7-client-go/internal/websocketx"
+	"github.com/m-lab/ndt7-client-go/mlabns"
+	"github.com/m-lab/ndt7-client-go/spec"
+)
+
+// LocateFn is the type of function used to locate a server.
+type LocateFn = func(client *mlabns.Client) (string, error)
+
+// connectFn is the type of the function used to create
+// a new *websocket.Conn connection.
+type connectFn = func(
+	dialer websocket.Dialer,
+	ctx context.Context, urlStr string,
+	requestHeader http.Header,
+) (*websocket.Conn, *http.Response, error)
+
+// subtestFn is the type of the function running a subtest.
+type subtestFn = func(
+	ctx context.Context, conn websocketx.Conn, ch chan<- spec.Measurement,
+)
+
+// defaultUserAgent is the default user agent used by this client.
+const defaultUserAgent = "ndt7-client-go/0.1.0"
+
+// Client is a ndt7 client.
+type Client struct {
+	// Ctx is the client context. It's set using NewClient.
+	Ctx context.Context
```

Comment:
Same comment as for the other context inside a struct.

Reply:
Fixed!